net/proxy_resolution/proxy_config_service_tor.cc

```diff
@@ -98,6 +107,14 @@ bool IsTorProxyConfig(const ProxyConfigWithAnnotation& config) {
          kTorProxyTrafficAnnotation.unique_id_hash_code;
 }
 
+std::string AnonymizationKeyToString(const NetworkAnonymizationKey& key) {
+  const std::optional<net::SchemefulSite>& schemeful_site =
+      key.GetTopFrameSite();
+  DCHECK(schemeful_site.has_value());
```

Comment:
I realize you're just moving existing code around, but should this be a `CHECK`? It seems like a failure here would lead to multiple sites sharing an empty host key.

Reply:
It will be undefined behavior if this failed when `schemeful_site->Serialize()`.
It's likely lead to a crash instead of empty key being shared but I agree it is safer to use `CHECK` here.

Reply:
Fixed and squashed